### PR TITLE
 #170947568 implements notifying a user for posted comment

### PR DESCRIPTION
**Pull request by bbaime98**
#### What does this PR do?
- add a send notification for posted comment feature
#### Description of Task to be completed?
- add a send notification function in comment controller
- add a send email notification function to comment controller
#### How should this be manually tested?
- create an account if you don't have one via `/api/v1/auth/register'` as requester
- login as a manager `/api/v1/auth/login'`
- post a comment to a respective request that was created by one of your requesters
- then that logged in requester will be notified
#### What are the relevant pivotal tracker stories?
[#170947568](https://www.pivotaltracker.com/story/show/170947568)
#### Screenshots (if appropriate)
image
image

#### Questions:

### CODE REVIEW

File: src/controllers/commentController.js [hunk 1]
```diff
@@ -20,11 +22,33 @@ export default class CommentController {
...
       const newComment = await db.Comments.create({
         requestId,
         commmentOwner: user.id,
         comment,
       });
+      if (user.role === 'manager') {
+        const notifyUser = await notifService.createNotif(req.request.userId, req.request.email, 'your manager posted a comment', '#');
+        const content = {
+          intro: req.__(notifyUser.content),
+          instruction: req.__(`To view this ${newComment.comment} click below`),
```
Comment (izzett222) [inline src/controllers/commentController.js]: could add internalization for these strings

File: src/controllers/commentController.js [hunk 2]
```diff
@@ -20,11 +22,33 @@ export default class CommentController {
...
+          text: req.__('View comment'),
+          signature: req.__('signature')
+        };
+        // await
+        SendNotification.SendNotif(notifyUser, req, content);
```
Comment (izzett222) [inline src/controllers/commentController.js]: hey @bbaime98 could you change the SendNotif function to lowercase because it's not a class

**Review by izzett222 — CHANGES_REQUESTED** [review]

File: src/controllers/commentController.js (hunk 2 above)
Comment (bbaime98) [reply src/controllers/commentController.js]: okay let me do it.

**Review by Baraka-Mugisha — DISMISSED** [review]

File: src/tests/comments.test.js [hunk 3]
```diff
@@ -1,17 +1,48 @@
...
 chai.use(chaiHttp);
 
 describe('COMMENTS TESTS', () => {
+  const socketToken = provideToken('79660e6f-4b7d-4g21-81re-74f54e9e1c8a', true, 'jeanne@andela.com', 'requester');
+  let clientSocket;
+  const BASE_URL = process.env.BASE_URL || `http://localhost:${server.address().port}`;
+  beforeEach((done) => {
+    sinon.stub(sgMail, 'send').resolves({
+      to: 'jeanaime@andela.com',
+      from: 'devrepublic.team@gmail.com',
```
Comment (jabichris) [inline src/tests/comments.test.js:27]: Try not to use real emails!

File: src/services/localesServices/locales/en.json [hunk 4]
```diff
@@ -125,5 +124,22 @@
...
+	"As a your manager I can add any comment on your request. I can also cancel or approve your request but let me comment first": "As a your manager I can add any comment on your request. I can also cancel or approve your request but let me comment first",
+	"thanks for your comment manager. This is my request too. And please Approve my request after this comment": "thanks for your comment manager. This is my request too. And please Approve my request after this comment",
+	"your manager posted a comment": "your manager posted a comment",
+	"your requester posted a comment": "your requester posted a comment",
+	"To view this yeahahaha 252525626262 click below": "To view this yeahahaha 252525626262 click below",
+	"null value in column \"managerId\" violates not-null constraint": "null value in column \"managerId\" violates not-null constraint",
+	"To view this thanks for your comment manager. This is my request too. And please Approve my request after this comment click below": "To view this thanks for your comment manager. This is my request too. And please Approve my request after this comment click below",
```
Comment (Baraka-Mugisha) [inline src/services/localesServices/locales/en.json]: could you remove comments that are not used. I think they may have been generated when testing.

File: src/seeders/20200212121323-User.js [hunk 5]
```diff
@@ -141,6 +141,18 @@ module.exports = {
         createdAt: new Date(),
         updatedAt: new Date(),
       },
+      {
+        id: 'a3426db1-1d2f-4c2c-8f25-c99962aff3fe',
+        firstName: 'bien',
+        lastName: 'aime',
+        email: 'jean@okay.com',
```
Comment (Baraka-Mugisha) [inline src/seeders/20200212121323-User.js:148]: This looks good!

File: src/controllers/commentController.js [hunk 6]
```diff
@@ -20,11 +22,31 @@ export default class CommentController {
...
+          instruction: req.__('Your Manager commented: %s', newComment.comment),
+          text: req.__('View comment'),
+          signature: req.__('signature')
+        };
+        await SendNotification.sendNotif(notifyUser, req, content);
+      } else {
+        const notifyUser = await notifService.createNotif(req.request.managerId, email, 'your requester posted a comment', '#');
+        const content = {
+          intro: req.__(notifyUser.content),
+          instruction: req.__('Your requsester commented: %s', newComment.comment),
```
Comment (Baraka-Mugisha) [inline src/controllers/commentController.js]: change 'requsester' to 'requester'

Comment (bbaime98) [reply src/controllers/commentController.js]: oh yeah

File: src/services/localesServices/locales/en.json (hunk 4 above)
Comment (bbaime98) [reply src/services/localesServices/locales/en.json]: sure

Comment (bbaime98) [reply src/services/localesServices/locales/en.json]: done

File: src/controllers/commentController.js (hunk 1 above)
Comment (bbaime98) [reply src/controllers/commentController.js]: it is done.

File: src/controllers/commentController.js (hunk 2 above)
Comment (bbaime98) [reply src/controllers/commentController.js]: @izzett222  it has been implemented.

**Review by izzett222 — APPROVED** [review]

**Review by Baraka-Mugisha — APPROVED** [review]

**Review by Ntare22 — APPROVED** [review]

**Pull request merged**